pandas/core/groupby/groupby.py

```diff
@@ -2109,7 +2119,12 @@ def cumcount(self, ascending: bool = True):
         with _group_selection_context(self):
             index = self._selected_obj.index
             cumcounts = self._cumcount_array(ascending=ascending)
-            return Series(cumcounts, index)
+
+            # GH28330 preserve subclassed Series/DataFrames through calls
```

Comment:
why is this needed?

Reply:
This was the same logic as for `GroupBy.ngroup()`

Reply:
I would actually leave these. I don't think returning a subclass makes any sense here.

Reply:
I reverted these changes -- `ngroup()` and `cumcount()` -- in the most recent set of commits.

Reply:
Thinking about this a bit more -- I'm not sure if it makes sense to have certain groupby-agg operations return the parent `Series` rather than a subclass. It seems more consistent to me to have all groupby.agg and groupby.transform preserve subclasses.